### PR TITLE
ebpf: reducing ebpf call overhead by using sampling instead of tracing every calls

### PR DESCRIPTION
**Pull request by rootfs**
replacing #823

### CODE REVIEW

**Comment by rootfs** [comment]
@marceloamaral @sunya-ch I mark the config as experimental. Can we first merge this PR to allow us to experiment with the config and find the optimal sampling rate later?

**Review by sunya-ch — APPROVED** [review]
/lgtm 
Just suggest to set default the same for both bcc and libbpf.

File: bpfassets/libbpf/src/kepler.bpf.c [hunk 1]
```diff
@@ -44,6 +44,9 @@ BPF_ARRAY(cache_miss, u64);
 // cpu freq counters
 BPF_ARRAY(cpu_freq_array, u32);
 
+int sample_rate = 1000;
```
[the remark this answers is not in the document]
Comment (rootfs) [reply bpfassets/libbpf/src/kepler.bpf.c]: i set them to 1 for now, the module can be loaded.

[the remark this answers is not in the document]
Comment (sunya-ch) [reply bpfassets/libbpf/src/kepler.bpf.c]: I see. Thank you.

File: bpfassets/perf_event/perf_event.c [hunk 2]
```diff
@@ -192,6 +197,19 @@ static inline u64 get_on_cpu_avg_freq(u32 *cpu_id, u64 on_cpu_cycles_delta, u64
 // int kprobe__finish_task_switch(switch_args *ctx)
 int kprobe__finish_task_switch(struct pt_regs *ctx, struct task_struct *prev)
 {
+    u32 initial = SAMPLE_RATE, *sample_counter_value, sample_counter_key = 1234;
```
Comment (marceloamaral) [inline bpfassets/perf_event/perf_event.c:200]: Since the sample rate is 0, it should be fine for now!

**Pull request merged**